app/views/ChangeAvatarView/index.tsx

```diff
@@ -0,0 +1,237 @@
+	payload?: Partial<IState>;
+}
+
+interface IState extends IAvatar {
+	resetUserAvatar: string;
+}
+
+const initialState = {
+	data: '',
+	url: '',
+	contentType: '',
+	service: '',
+	resetUserAvatar: ''
+};
+
+function reducer(state: IState, action: IReducerAction) {
+	const { type, payload } = action;
+	if (type in AvatarStateActions) {
+		return {
+			...initialState,
+			...payload
+		};
+	}
+	return state;
+}
+
+const ChangeAvatarView = () => {
+	const [state, dispatch] = useReducer(reducer, initialState);
+	const [saving, setSaving] = useState(false);
+	const { colors } = useTheme();
+	const { userId, username } = useAppSelector(
+		state => ({
+			userId: getUserSelector(state).id,
+			username: getUserSelector(state).username
+		}),
+		shallowEqual
+	);
+	const isDirty = useRef<boolean>(false);
+	const navigation = useNavigation<StackNavigationProp<ChatsStackParamList, 'ChangeAvatarView'>>();
+	const { context, titleHeader, room, t } = useRoute<RouteProp<ChatsStackParamList, 'ChangeAvatarView'>>().params;
+
+	useLayoutEffect(() => {
+		navigation.setOptions({
+			title: titleHeader || I18n.t('Avatar')
+		});
+	}, [titleHeader, navigation]);
+
+	useEffect(() => {
+		navigation.addListener('beforeRemove', e => {
+			if (!isDirty.current) {
+				return;
+			}
+
+			e.preventDefault();
+
+			showConfirmationAlert({
+				title: I18n.t('Discard_changes'),
+				message: I18n.t('Discard_changes_description'),
+				confirmationText: I18n.t('Discard'),
+				onPress: () => {
+					navigation.dispatch(e.data.action);
+				}
+			});
+		});
+	}, [navigation]);
+
+	const dispatchAvatar = (action: IReducerAction) => {
+		isDirty.current = true;
+		dispatch(action);
+	};
+
+	const submit = async () => {
+		try {
+			setSaving(true);
+			if (context === 'room' && room?.rid) {
+				// Change Rooms Avatar
+				await changeRoomsAvatar(room.rid, state?.data);
+			} else if (state?.url) {
+				// Change User's Avatar
+				await changeUserAvatar(state);
+			} else if (state.resetUserAvatar) {
+				// Change User's Avatar
+				await resetUserAvatar(userId);
+			}
+			isDirty.current = false;
+		} catch (e: any) {
+			log(e);
+			return showErrorAlert(e.message, I18n.t('Oops'));
+		} finally {
+			setSaving(false);
+		}
+		return navigation.goBack();
+	};
+
+	const pickImage = async () => {
+		const options = {
+			cropping: true,
+			compressImageQuality: 0.8,
+			freeStyleCropEnabled: true,
+			cropperAvoidEmptySpaceAroundImage: false,
+			cropperChooseText: I18n.t('Choose'),
+			cropperCancelText: I18n.t('Cancel'),
+			includeBase64: true
+		};
+		try {
+			const response: Image = await ImagePicker.openPicker(options);
+			dispatchAvatar({
+				type: AvatarStateActions.CHANGE_AVATAR,
+				payload: { url: response.path, data: `data:image/jpeg;base64,${response.data}`, service: 'upload' }
+			});
+		} catch (error) {
+			log(error);
+		}
+	};
+
+	const deletingRoomAvatar = context === 'room' && state.data === null ? {} : { rid: room?.rid };
```

Comment:
We talked about using Avatar component rather than container to remove this logic. What changed?

I'm missing a clear reasoning from you about what was discussed on your PRs.
If it can't be done, please say so as a note for the reviewer.
Otherwise it looks like I'm just wasting time repeating same reviews over and over.

Reply:
Yeap, sorry, that was missing explain why I didn't use the AvatarPresentational. So I didn't use it because I needed to get another param inside this View, for example, the `server`, this param is passed from  AvatarContainer passes to Avatar

Reply:
I changed and pushed a new commit with this change too